app/services/supplementary_data.py

```diff
@@ -78,6 +91,24 @@ def get_supplementary_data(*, dataset_id: str, unit_id: str, survey_id: str) ->
     raise SupplementaryDataRequestFailed
 
 
+def decrypt_supplementary_data(supplementary_data: MutableMapping) -> Mapping:
+    if (encryption_key_id := supplementary_data.get("encryption_key_id")) and (
+        encrypted_data := supplementary_data.get("data")
+    ):
+        key = (
+            get_key_store()
+            .get_private_key_by_kid(KEY_PURPOSE_SDS, encryption_key_id)
+            .as_jwk()
```

Comment:
If the above is done, we can just use the `decrypt` method.

Reply:
Done 👍